### ISSUE TITLE
Livekit server can't start because of "invalid" TURN configurations

### ISSUE THREAD

**Issue by bmbferreira** [livekit/livekit#1098]
**Describe the bug**
Because of the config validation introduced on [this PR](https://github.com/livekit/livekit/pull/1051) the livekit server now can't start on the latest version because of TURN configurations that are set on the helm chart (such as the loadBalancer annotations). It stays in crashloopbackoff with the following logs:
```
could not parse config: yaml: unmarshal errors:
  line 15: field loadBalancerAnnotations not found in type config.TURNConfig
  line 17: field secretName not found in type config.TURNConfig
```

And this is what I'm setting on the helm chart values:
```
autoscaling:
  enabled: true
  maxReplicas: 3
  minReplicas: 1
  targetCPUUtilizationPercentage: 60
livekit:
  keys:
    asdasdasdasd123: ASDASDasDASDASDASD
  redis:
    address: my-redis.abcdef.ng.0003.euw1.cache.amazonaws.com:6666
  rtc:
    use_external_ip: true
  turn:
    domain: mytest4312.dev
    enabled: true
    loadBalancerAnnotations:
      external-dns.alpha.kubernetes.io/hostname: turn.live.mytest4312.dev
    secretName: livekit-turnz
    tls_port: 3478
  webhook:
    api_key: asdasdasdasd123
    urls:
    - https://rlyd.dev/api/live/events
loadBalancer:
  tls:
  - hosts:
    - live.mytest4312.dev
  type: alb
nodeSelector:
  eks.amazonaws.com/nodegroup: livekit-server-workerz
replicaCount: 2
resources:
  limits:
    memory: 2048Mi
  requests:
    cpu: 7000m
    memory: 1024Mi
```
That sets this on the envirorment variable with the config file inside the pod (`kubectl exec -it livekit-server-5fdd96f9b6-d42nz  -- env`):
```
...
LIVEKIT_CONFIG=keys:
...
turn:
  domain: turn.live.mytest4312.dev
  enabled: true
  loadBalancerAnnotations:
    external-dns.alpha.kubernetes.io/hostname: turn.live.mytest4312.dev
  secretName: livekit-turnz
  tls_port: 3478
...
```

**Server**
- Version: 1.2.4
- Environment: EKS

**To Reproduce**
1. Update livekit server to latest version (1.2.4)
2. Set a secret and loadBalancerAnnotations for TURN on the helm chart.

**Expected behavior**
If these configurations on the helm chart are not valid to be parsed inside the livekit-server then they shouldn't be passed to the envirornment variable tat contains the configuration file to be parsed by the server.

**Comment by bmbferreira** [comment]
One more thing is that the main problem might be here on the helm chart: https://github.com/livekit/livekit-helm/blob/master/livekit-server/templates/configmap.yaml#L7

It is setting on the configmap all the values "blindly", including the ones that the server won't be able to parse as a config (in my case `loadBalancerAnnotations` and `secretName` but maybe there's other configs that will also trigger this issue).

**Comment by davidzhao** [comment]
ugh.. you are right this is a problem. I think in the near term, we can introduce a mode where it bypasses strict mode. I'll take a look at this tomorrow.

**Comment by bmbferreira** [comment]
Thanks @davidzhao! yup, that could be a solution. Keeping the logs in non-strict mode would be helpful anyways 🤷‍♂️ 

Another solution could be on the helm chart avoid passing "invalid" configurations to the configmap (in this case `loadBalancerAnnotations` and `secretName`) It would get more verbose though...

**Comment by davidzhao** [comment]
This was fixed in #1105 and released in v1.2.5. Give it a shot and let me know how this works for you!.

**Comment by bmbferreira** [comment]
thank you @davidzhao 🙇
